rust/models/src/credential.rs

```diff
@@ -114,6 +127,11 @@ pub enum CredentialType {
         #[cfg_attr(feature = "serde_support", serde(rename = "private"))]
         /// The private key for authentication.
         private_key: String,
+        /// privilege credential only use for SSH service
+        #[serde(skip_serializing_if = "Option::is_none")]
+        #[serde(default)]
```

Comment:
I think we have to use `#[cfg_attr(feature = "serde_support"` everywhere when we want to use serde. Otherwise compilation fails when serde is not available. Alternatively we mark serde as required in the cargo and drop the compiler declaration for cfg_attr.

Luckily you can combine them:
```
#[cfg_attr(feature = "serde_support", serde(default, flatten, skip_serializing_if = "Option::is_none"))]
```

should be valid.